ledger/metrics/src/main/scala/com/daml/metrics/LAPIMetrics.scala

```diff
@@ -7,7 +7,9 @@ import com.daml.metrics.api.MetricDoc.MetricQualification.{Debug, Traffic}
 import com.daml.metrics.api.MetricHandle.{Counter, MetricsFactory}
 import com.daml.metrics.api.{MetricDoc, MetricHandle, MetricName, MetricsContext}
 
-class LAPIMetrics(val prefix: MetricName, val factory: MetricsFactory) {
+import scala.annotation.nowarn
+
+class LAPIMetrics(val prefix: MetricName, @deprecated @nowarn val factory: MetricsFactory) {
```

Comment:
Any specific logic why values defined with the type `MetricsFactory` are sometime marked `@deprecated @nowarn`, and sometime only `@nowarn` ?

Reply:
When the `metricFactory` is just passed along then I added only `@nowarn`, if it's used to actually create metrics then I also added the `@deprecated`. In some cases I did not add the deprecated because there were simply too many usages.